primitives/src/xdns.rs

```diff
@@ -86,6 +88,9 @@ pub struct TokenRecord {
 
     /// Token properties - decimals, symbol, name
     pub token_props: TokenInfo,
+
+    /// Token location, assume Option<MultiLocation>
+    pub token_location: Option<MultiLocation>,
```

Comment:
how about moving token_location as 3rd option to TokenInfo enum? You wouldn't need to update CLI with new parameter while register_gateway

Reply:
Yeah, I was thinking about it... However, the asset will have a **different** `MultiLocation` depending on whether it is **foreign** or **local** for a parachain or whether it is an asset from a different ecosystem(could ignore this one for now). If we want to record the `MultiLocation` as `TokenInfo` we could refer to a single list of `MultiLocations` for the asset though.

Reply:
got it, then please fix the CI client for current registration, so that we won't loose that functionality after merging this

Reply:
fixed the gateway registration + updated @t3rn/types in [547f701](https://github.com/t3rn/t3rn/pull/1469/commits/547f701a09f18f7a0aca24ca08b45de5146c22ce)